Parse latest XSD version of OWASP Dependency-Check

Parsing a latest dep check report gives the following error:

```
Traceback (most recent call last):
  File "/home/collector/base_collectors/source_collector.py", line 130, in __safely_parse_source_responses
    measurement = await self._parse_source_responses(responses)
  File "/home/collector/source_collectors/file_source_collectors/owasp_dependency_check.py", line 30, in _parse_source_responses
    tree, namespaces = await parse_source_response_xml_with_namespace(response, self.allowed_root_tags)
  File "/home/collector/collector_utilities/functions.py", line 27, in parse_source_response_xml_with_namespace
    tree = await parse_source_response_xml(response, allowed_root_tags)
  File "/home/collector/collector_utilities/functions.py", line 20, in parse_source_response_xml
    raise AssertionError(f'The XML root element should be one of "{allowed_root_tags}" but is "{tree.tag}"')
AssertionError: The XML root element should be one of "['{https://jeremylong.github.io/DependencyCheck/dependency-check.2.0.xsd}analysis', '{https://jeremylong.github.io/DependencyCheck/dependency-check.2.1.xsd}analysis', '{https://jeremylong.github.io/DependencyCheck/dependency-check.2.2.xsd}analysis', '{https://jeremylong.github.io/DependencyCheck/dependency-check.2.3.xsd}analysis']" but is "{https://jeremylong.github.io/DependencyCheck/dependency-check.2.5.xsd}analysis"
```

components\collector\src\source_collectors\file_source_collectors\owasp_dependency_check.py:19

```    allowed_root_tags = [f"{{https://jeremylong.github.io/DependencyCheck/dependency-check.{version}.xsd}}analysis"
                         for version in ("2.0", "2.1", "2.2", "2.3")] ```

Should allow XSD version 2.5 but is there any reason why we should not allow any XSD version ?

--- a/components/collector/src/source_collectors/file_source_collectors/owasp_dependency_check.py
+++ b/components/collector/src/source_collectors/file_source_collectors/owasp_dependency_check.py
@@ -17,7 +17,7 @@ class OWASPDependencyCheckBase(XMLFileSourceCollector, ABC):  # pylint: disable=
     """Base class for OWASP Dependency Check collectors."""
 
     allowed_root_tags = [f"{{https://jeremylong.github.io/DependencyCheck/dependency-check.{version}.xsd}}analysis"
-                         for version in ("2.0", "2.1", "2.2", "2.3")]
+                         for version in ("2.0", "2.1", "2.2", "2.3", "2.4", "2.5")]
 
 
 class OWASPDependencyCheckDependencies(OWASPDependencyCheckBase):
@@ -42,7 +42,7 @@ class OWASPDependencyCheckDependencies(OWASPDependencyCheckBase):
         """Parse the entity from the dependency."""
         file_path = dependency.findtext("ns:filePath", default="", namespaces=namespaces)
         sha1 = dependency.findtext("ns:sha1", namespaces=namespaces)
-        # We can only generate a entity landing url if a sha1 is present in the XML, but unfortunately not all
+        # We can only generate an entity landing url if a sha1 is present in the XML, but unfortunately not all
         # dependencies have one, so check for it:
         entity_landing_url = f"{landing_url}#l{dependency_index + 1}_{sha1}" if sha1 else ""
         key = sha1 if sha1 else sha1_hash(file_path)

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -9,6 +9,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 <!-- The line "## <square-bracket>Unreleased</square-bracket>" is replaced by the ci/release.py script with the new release version and release date. -->
 
+## [Unreleased]
+
+### Added
+
+- Support version 2.4 and 2.5 of the OWASP Dependency Check XSD. Closes [#1460](https://github.com/ICTU/quality-time/issues/1460).
+
 ## [3.5.0] - [2020-09-12]
 
 ### Fixed

